[DOC] doctest MEGA-issue

## Expected behavior ##

<!-- A clear and concise description of what you want to do and what you think should happen. (Code to reproduce the behavior can be added below). -->

All the code in our docstring examples should be self-contained and run on its own. They should:

- import the correct modules
- demonstrate the desired behaviour
- run correctly

## Actual behavior ##

A lot of the examples in our docs do not run correctly as they are missing imports, steps, some code logic or all of the above. 

By my initial count there are ~350 failing examples and only a handful that work.

We would love this to be fixed! **This is a great opportunity for community members and budding open source contributors to get involved as there are lots of docstrings to tackle.**

Luckily there is a way to test which doc examples are failing  programatically!

1. Navigate to the `/package/doc/sphinx`  directory
2. run `make doctest`

This will indicate which doc examples are failing. 

If you want to tackle multiple doc changes at once  editing your makefile command on L130 to include `--keep-going`  will allow you to see the full set of errors. 

eg 
```
#L130
	$(SPHINXBUILD) -b doctest $(ALLSPHINXOPTS) $(BUILDDIR)/doctest --keep-going
```

**Please contribute your changes in a new PR!**

If you are new to making PRs and github in general, try our handy [contributing guide](https://userguide.mdanalysis.org/stable/contributing.html)! Feel free to ping our team if unsure. :)

- currently i am working on **cluster.py** files doctest error (it contains five errors, from which i have solved 4, but stuck at one).
I am not able to solve this error, can you please guide me in this ?

**code**: [cluster.py](https://github.com/MDAnalysis/mdanalysis/blob/2d1d6845134ec24279af6f35b1f5ddd1d77edd11/package/MDAnalysis/analysis/encore/clustering/cluster.py#L147)

<br>

**Error**:
```bash
UNEXPECTED EXCEPTION: AttributeError("'ClusterCollection' object has no attribute 'metadata'")
Traceback (most recent call last):
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/doctest.py", line 1350, in __run
    exec(compile(example.source, filename, "single",
  File "<doctest MDAnalysis.analysis.encore.clustering.cluster.cluster[10]>", line 1, in <module>
  File "<doctest MDAnalysis.analysis.encore.clustering.cluster.cluster[10]>", line 1, in <listcomp>
AttributeError: 'ClusterCollection' object has no attribute 'metadata'
/home/heet/Work/Development/mdanalysis/package/MDAnalysis/analysis/encore/clustering/cluster.py:145: UnexpectedException
===================================================== short test summary info =====================================================
FAILED analysis/encore/clustering/cluster.py::MDAnalysis.analysis.encore.clustering.cluster.cluster
```

```
(mdanalysis-dev) (base) lunamorrow@d-i89-211-193 sphinx % make doctest                                                
sphinx-build -v -W -b doctest   source ../doctest --keep-going
Running Sphinx v4.2.0

Traceback (most recent call last):
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/config.py", line 328, in eval_config_file
    exec(code, namespace)
  File "/Users/lunamorrow/mdanalysis/package/doc/sphinx/source/conf.py", line 17, in <module>
    import MDAnalysis as mda
ModuleNotFoundError: No module named 'MDAnalysis'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/cmd/build.py", line 276, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/application.py", line 216, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/config.py", line 172, in read
    namespace = eval_config_file(filename, tags)
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/config.py", line 341, in eval_config_file
    raise ConfigError(msg % traceback.format_exc()) from exc
sphinx.errors.ConfigError: There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/config.py", line 328, in eval_config_file
    exec(code, namespace)
  File "/Users/lunamorrow/mdanalysis/package/doc/sphinx/source/conf.py", line 17, in <module>
    import MDAnalysis as mda
ModuleNotFoundError: No module named 'MDAnalysis'


Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/Users/lunamorrow/opt/anaconda3/lib/python3.9/site-packages/sphinx/config.py", line 328, in eval_config_file
    exec(code, namespace)
  File "/Users/lunamorrow/mdanalysis/package/doc/sphinx/source/conf.py", line 17, in <module>
    import MDAnalysis as mda
ModuleNotFoundError: No module named 'MDAnalysis'

make: *** [doctest] Error 2

```
Ok noted, thanks @hmacdope, I will look for another issue to tackle. I saw a few tagged with GSOC starter that did not already have a pull submitted, but it seems someone has already requested to do them for their GSOC PR and I did not want to double up as I understand contributions/pull requests are on a first submitted basis. I will have another look :)

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -68,40 +68,34 @@ LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
 ..testsetup::
 
-    import MDAnalysis
-
+    >>> import MDAnalysis
+    >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
 ..doctest::
 
-    >>> u = MDAnalysis.Universe("lammps.data", "lammps_real.dcd", format="LAMMPS")
+    >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
 
 If the trajectory uses *units nano* then use ::
 
-.. testsetup::
-
-    import MDAnalysis
-
 .. doctest::
 
-    >>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", format="LAMMPS",
-    ...                          lengthunit="nm", timeunit="ns")
+   >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
+   ...                          lengthunit="nm", timeunit="ns")
 
 To scan through a trajectory to find a desirable frame and write to a LAMMPS
 data file,
 
-.. testsetup::
-
-    import MDAnalysis
-
-.. doctest::
-
-    >>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", 
-    ...                             format="LAMMPS", lengthunit="nm", timeunit="ns")
-    >>> for ts in u.trajectory:
-    ...     # analyze frame
-    ...     if take_this_frame == True:
-    ...         with mda.Writer('frame.data') as W:
-    ...             W.write(u.atoms)
-    ...         break
+   >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
+   ...                          lengthunit="nm", timeunit="ns")
+   >>> take_this_frame = False
+   >>> for ts in u.trajectory:
+   ...     # analyze frame
+   ...     if ts.frame == 4:
+   ...         take_this_frame = True
+   ...     if take_this_frame == True:
+   ...         with MDAnalysis.Writer('frame.data') as W:
+   ...             W.write(u.atoms)
+   ...         break
+    
 
 Note
 ----

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -61,20 +61,20 @@ The DumpReader expects ascii dump files written with the default
 
 Example: Loading a LAMMPS simulation
 ------------------------------------
-..testsetup::
+.. testsetup::
 
 To load a LAMMPS simulation from a LAMMPS data file (using the
 :class:`~MDAnalysis.topology.LAMMPSParser.DATAParser`) together with a
 LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
-    >>> import MDAnalysis
-    >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
-    >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
+>>> import MDAnalysis
+>>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
+>>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
 
-If the trajectory uses *units nano* then use 
+If the trajectory uses *units nano* then use
 
-   >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
-   ...                          lengthunit="nm", timeunit="ns")
+>>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
+...                          lengthunit="nm", timeunit="ns")
 
 To scan through a trajectory to find a desirable frame and write to a LAMMPS
 data file,
@@ -89,7 +89,7 @@ data file,
    ...     if take_this_frame == True:
    ...         with MDAnalysis.Writer('frame.data') as W:
    ...             W.write(u.atoms)
-   ...         break    
+   ...         break
 
 Note
 ----

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -66,9 +66,21 @@ To load a LAMMPS simulation from a LAMMPS data file (using the
 :class:`~MDAnalysis.topology.LAMMPSParser.DATAParser`) together with a
 LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
+..testsetup:: 
+    
+    import MDAnalysis
+
+..doctest::    
+
     >>> u = MDAnalysis.Universe("lammps.data", "lammps_real.dcd", format="LAMMPS")
 
 If the trajectory uses *units nano* then use ::
+
+.. testsetup:: 
+    
+    import MDAnalysis
+
+.. doctest::
 
     >>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", format="LAMMPS",
     ...                          lengthunit="nm", timeunit="ns")
@@ -76,6 +88,14 @@ If the trajectory uses *units nano* then use ::
 To scan through a trajectory to find a desirable frame and write to a LAMMPS
 data file,
 
+.. testsetup:: 
+    
+    import MDAnalysis
+
+.. doctest::
+
+>>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", format="LAMMPS",
+...                          lengthunit="nm", timeunit="ns")
 >>> for ts in u.trajectory:
 ...     # analyze frame
 ...     if take_this_frame == True:

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -61,22 +61,21 @@ The DumpReader expects ascii dump files written with the default
 
 Example: Loading a LAMMPS simulation
 ------------------------------------
-.. testsetup::
 
 To load a LAMMPS simulation from a LAMMPS data file (using the
 :class:`~MDAnalysis.topology.LAMMPSParser.DATAParser`) together with a
 LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
->>> import MDAnalysis
->>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
->>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
+    >>> import MDAnalysis
+    >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
+    >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
 
 If the trajectory uses *units nano* then use
 
->>> import MDAnalysis
->>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
->>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
-...                          lengthunit="nm", timeunit="ns")
+    >>> import MDAnalysis
+    >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
+    >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
+    ...                          lengthunit="nm", timeunit="ns")
 
 To scan through a trajectory to find a desirable frame and write to a LAMMPS
 data file,

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -66,18 +66,18 @@ To load a LAMMPS simulation from a LAMMPS data file (using the
 :class:`~MDAnalysis.topology.LAMMPSParser.DATAParser`) together with a
 LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
-..testsetup:: 
-    
+..testsetup::
+
     import MDAnalysis
 
-..doctest::    
+..doctest::
 
     >>> u = MDAnalysis.Universe("lammps.data", "lammps_real.dcd", format="LAMMPS")
 
 If the trajectory uses *units nano* then use ::
 
-.. testsetup:: 
-    
+.. testsetup::
+
     import MDAnalysis
 
 .. doctest::
@@ -94,14 +94,14 @@ data file,
 
 .. doctest::
 
->>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", format="LAMMPS",
-...                          lengthunit="nm", timeunit="ns")
->>> for ts in u.trajectory:
-...     # analyze frame
-...     if take_this_frame == True:
-...         with mda.Writer('frame.data') as W:
-...             W.write(u.atoms)
-...         break
+    >>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", format="LAMMPS",
+    ...                          lengthunit="nm", timeunit="ns")
+    >>> for ts in u.trajectory:
+    ...     # analyze frame
+    ...     if take_this_frame == True:
+    ...         with mda.Writer('frame.data') as W:
+    ...             W.write(u.atoms)
+    ...         break
 
 Note
 ----

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -61,23 +61,17 @@ The DumpReader expects ascii dump files written with the default
 
 Example: Loading a LAMMPS simulation
 ------------------------------------
+..testsetup::
 
 To load a LAMMPS simulation from a LAMMPS data file (using the
 :class:`~MDAnalysis.topology.LAMMPSParser.DATAParser`) together with a
 LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
-..testsetup::
-
     >>> import MDAnalysis
     >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
-
-..doctest::
-
     >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
 
-If the trajectory uses *units nano* then use ::
-
-.. doctest::
+If the trajectory uses *units nano* then use 
 
    >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
    ...                          lengthunit="nm", timeunit="ns")

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -70,6 +70,7 @@ LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
     >>> import MDAnalysis
     >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
+
 ..doctest::
 
     >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS")
@@ -94,8 +95,7 @@ data file,
    ...     if take_this_frame == True:
    ...         with MDAnalysis.Writer('frame.data') as W:
    ...             W.write(u.atoms)
-   ...         break
-    
+   ...         break    
 
 Note
 ----

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -88,14 +88,14 @@ If the trajectory uses *units nano* then use ::
 To scan through a trajectory to find a desirable frame and write to a LAMMPS
 data file,
 
-.. testsetup:: 
-    
+.. testsetup::
+
     import MDAnalysis
 
 .. doctest::
 
-    >>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", format="LAMMPS",
-    ...                          lengthunit="nm", timeunit="ns")
+    >>> u = MDAnalysis.Universe("lammps.data", "lammps_nano.dcd", 
+    ...                             format="LAMMPS", lengthunit="nm", timeunit="ns")
     >>> for ts in u.trajectory:
     ...     # analyze frame
     ...     if take_this_frame == True:

--- a/package/MDAnalysis/coordinates/LAMMPS.py
+++ b/package/MDAnalysis/coordinates/LAMMPS.py
@@ -73,12 +73,16 @@ LAMMPS DCD with "*real*" provide the keyword *format="LAMMPS*"::
 
 If the trajectory uses *units nano* then use
 
+>>> import MDAnalysis
+>>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2
 >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
 ...                          lengthunit="nm", timeunit="ns")
 
 To scan through a trajectory to find a desirable frame and write to a LAMMPS
 data file,
 
+   >>> import MDAnalysis
+   >>> from MDAnalysis.tests.datafiles import LAMMPSdata2, LAMMPSdcd2 
    >>> u = MDAnalysis.Universe(LAMMPSdata2, LAMMPSdcd2, format="LAMMPS",
    ...                          lengthunit="nm", timeunit="ns")
    >>> take_this_frame = False

